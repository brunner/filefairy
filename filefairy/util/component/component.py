#!/usr/bin/env python3
# -*- coding: utf-8 -*-


def anchor(url, content):
    return '<a href="{}">{}</a>'.format(url, content)


def card(href='', title='', info='', table=None, ts='', success='', danger=''):
    return {
        'href': href,
        'title': title,
        'info': info,
        'table': table,
        'ts': ts,
        'success': success,
        'danger': danger
    }


def span(classes, text):
    return '<span class="{}">{}</span>'.format(' '.join(classes), text)


def table(clazz='border mt-3', hcols=None, bcols=None, head=None, body=None):
    return {
        'clazz': clazz,
        'hcols': hcols,
        'bcols': bcols,
        'head': head,
        'body': body
    }
