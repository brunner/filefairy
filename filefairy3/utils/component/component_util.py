#!/usr/bin/env python


def card(href='',
         title='',
         info='',
         table=[],
         ts='',
         success='',
         danger=''):
    return {
        'href': href,
        'title': title,
        'info': info,
        'table': table,
        'ts': ts,
        'success': success,
        'danger': danger
    }
