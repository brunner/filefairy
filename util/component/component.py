#!/usr/bin/env python3
# -*- coding: utf-8 -*-


def anchor(url, content):
    return '<a href="{}">{}</a>'.format(url, content)


def bold(text):
    return span(['font-weight-bold'], text)


def card(href='',
         title='',
         info='',
         code='',
         table=None,
         ts='',
         success='',
         danger=''):
    return {
        'href': href,
        'title': title,
        'info': info,
        'code': code,
        'table': table,
        'ts': ts,
        'success': success,
        'danger': danger
    }


def cell(col=None, content=''):
    return {'col': col, 'content': content}


def col(clazz='', colspan=''):
    return {'clazz': clazz, 'colspan': colspan}


def show(trigger, text):
    s = '<a data-toggle="collapse" href="#{0}" role="button" ' + \
        'aria-expanded="false" aria-controls="{0}">{1}</a>'
    return s.format(trigger, text)


def replace(text, replace):
    s = '<a class="repl-toggler" href="#" role="button" data-repl="{}">{}</a>'
    return s.format(replace, text)


def secondary(text):
    return span(['text-secondary'], text)


def span(classes, text):
    return '<span class="{}">{}</span>'.format(' '.join(classes), text)


def table(clazz='border mt-3',
          id_='',
          hcols=None,
          bcols=None,
          fcols=None,
          head=None,
          body=None,
          foot=None):
    return {
        'clazz': clazz,
        'id': id_,
        'hcols': hcols,
        'bcols': bcols,
        'fcols': fcols,
        'head': head,
        'body': body,
        'foot': foot
    }
