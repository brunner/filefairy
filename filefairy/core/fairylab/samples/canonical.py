#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import os
import re
import sys

_path = os.path.dirname(os.path.abspath(__file__))
_root = re.sub(r'/core/fairylab/samples', '', _path)
sys.path.append(_root)
from util.component.component import card  # noqa

_breadcrumbs = [{'href': '', 'name': 'Home'}]

_browsable = [
    card(
        href='/fairylab/bar/',
        title='bar',
        info='Description of bar.',
        ts='10m ago',
    ),
    card(
        href='/fairylab/baz/',
        title='baz',
        info='Description of baz.',
        ts='1h ago',
        danger='failed'),
    card(
        href='/fairylab/foo/',
        title='foo',
        info='Description of foo.',
        ts='2m ago',
    )
]
_internal = [
    card(
        title='quux',
        info='Description of quux.',
        ts='15m ago',
        danger='failed'),
    card(
        title='quuz',
        info='Description of quuz.',
        ts='30s ago',
        success='just now',
    ),
    card(
        title='qux',
        info='Description of qux.',
        ts='2d ago',
    )
]

subtitle = ''

tmpl = 'home.html'

context = {
    'title': 'home',
    'breadcrumbs': _breadcrumbs,
    'browsable': _browsable,
    'internal': _internal,
}
