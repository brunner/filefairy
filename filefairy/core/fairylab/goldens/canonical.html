<!doctype html>
<html lang="en">

<head>

<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link rel="stylesheet"
      href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css"
      integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm"
      crossorigin="anonymous">
<link rel="stylesheet"
      href="https://cdnjs.cloudflare.com/ajax/libs/open-iconic/1.1.1/font/css/open-iconic-bootstrap.min.css">
<title>Fairylab | home</title>

<style>
html {
  position: relative;
  min-height: 100%;
}
body {
  margin-bottom: 60px;
}
header {
  background-color: #f5f5f5;
  width: 100%;
}
footer {
  background-color: #f5f5f5;
  bottom: 0;
  height: 60px;
  line-height: 60px;
  position: absolute;
  width: 100%;
}
.alert, .form-control {
  padding: .75rem;
}
.alert .oi {
  line-height: 24px;
  padding-right: .75rem;
}
.anchor {
  padding-top: 0px;
  margin-top: 0px;
  transition: padding-top .35s ease, margin-top .35s ease;
}
.breadcrumb {
  background-color: inherit;
  line-height: 60px;
}
.flex-grow {
  flex-grow: 1;
}
.left-8p {
  left: 8px;
}
.navbar {
  padding: 0;
}
.navbar-nav {
  padding: .5rem 0;
}
.navbar .nav-link {
  padding: .5rem;
}
.pl-24p {
  padding-left: 24px;
}
.pr-24p {
  padding-right: 24px;
}
.right-8p {
  right: 8px;
}
.table, .table-sm {
  margin-bottom: 0;
  word-break: break-word;
}
.table .thead-light td {
  background-color: #f5f5f5;
}
.table-fixed {
  table-layout: fixed;
}
.table-sm td {
  border-top: 0;
  padding: 0 .3rem 0 0;
}
.table .td-sm {
  padding: .75rem .25rem;
}
.top-14p {
  top: 14px;
}
.w-10 {
  width: 10%;
}
.w-20 {
  width: 20%;
}
.w-40 {
  width: 40%;
}
.w-55p {
  width: 55px;
}
.w-65p {
  width: 65px;
}
.w-75p {
  width: 75px;
}
.w-105p {
  width: 105px;
}
</style>
</head>

<body>

<!-- ^(base.header) -->
<header class="border-bottom sticky-top">
<div class="container">
<nav class="navbar navbar-light" aria-label="nav">
<ol class="breadcrumb m-0 p-0">
<li class="breadcrumb-item active" aria-current="page">Home</li>
</ol>

<!-- ^(common.nav) -->
<button class="navbar-toggler border-0" type="button" data-toggle="collapse"
        data-target="#nav-dropdown" aria-controls="#nav-dropdown"
        aria-expanded="false" aria-label="nav toggler">
<span class="navbar-toggler-icon"></span>
</button>
<div class="collapse navbar-collapse border-top" id="nav-dropdown">
<ul class="navbar-nav">
<li class="nav-item">
<a class="nav-link" href="#browsable">Browsable</a>
</li>
<li class="nav-item">
<a class="nav-link" href="#internal">Internal</a>
</li>
</ul>
</div>
<!-- $(common.nav) -->

</nav>
</div>
</header>
<!-- $(base.header) -->

<!-- ^(base.main) -->
<main role="main" class="pb-3">

<!-- ^(home.browsable) -->
<div class="container anchor" id="browsable">

<!-- ^(common.heading) -->
<div class="row">
<div class="col-12">
<div class="alert alert-secondary mt-3 mb-0">
<span class="oi oi-globe text-secondary"></span>Browsable
</div>
</div>
</div>
<!-- $(common.heading) -->

<!-- ^(home.row) -->
<div class="row d-flex flex-wrap">

<!-- ^(home.column) -->
<div class="col-md-6 col-lg-4 d-flex flex-column">

<!-- ^(common.card) -->
<div class="card d-flex flex-column flex-grow mt-3">

<!-- ^(common.card.body) -->
<div class="card-body d-flex flex-column">
<h5 class="card-title">
<a href="/fairylab/bar/">bar</a>
</h5>
<p class="card-text flex-grow">Description of bar.</p>
</div>
<!-- $(common.card.body) -->

<!-- ^(common.card.footer) -->
<div class="card-footer text-muted text-truncate">
<small>Last activity: 10m ago</small>
</div>
<!-- $(common.card.footer) -->

</div>
<!-- $(common.card) -->

</div>
<!-- $(home.column) -->

<!-- ^(home.column) -->
<div class="col-md-6 col-lg-4 d-flex flex-column">

<!-- ^(common.card) -->
<div class="card d-flex flex-column flex-grow mt-3">

<!-- ^(common.card.body) -->
<div class="card-body d-flex flex-column">
<h5 class="card-title">
<a href="/fairylab/baz/">baz</a>
</h5>
<p class="card-text flex-grow">Description of baz.</p>
</div>
<!-- $(common.card.body) -->

<!-- ^(common.card.footer) -->
<div class="card-footer text-muted text-truncate">
<small>Last activity: 1h ago<span class="text-danger"> (failed)</span></small>
</div>
<!-- $(common.card.footer) -->

</div>
<!-- $(common.card) -->

</div>
<!-- $(home.column) -->

<!-- ^(home.column) -->
<div class="col-md-6 col-lg-4 d-flex flex-column">

<!-- ^(common.card) -->
<div class="card d-flex flex-column flex-grow mt-3">

<!-- ^(common.card.body) -->
<div class="card-body d-flex flex-column">
<h5 class="card-title">
<a href="/fairylab/foo/">foo</a>
</h5>
<p class="card-text flex-grow">Description of foo.</p>
</div>
<!-- $(common.card.body) -->

<!-- ^(common.card.footer) -->
<div class="card-footer text-muted text-truncate">
<small>Last activity: 2m ago</small>
</div>
<!-- $(common.card.footer) -->

</div>
<!-- $(common.card) -->

</div>
<!-- $(home.column) -->

</div>
<!-- $(home.row) -->

</div>
<!-- $(home.browsable) -->

<!-- ^(home.internal) -->
<div class="container anchor" id="internal">

<!-- ^(common.heading) -->
<div class="row">
<div class="col-12">
<div class="alert alert-secondary mt-3 mb-0">
<span class="oi oi-shield text-secondary"></span>Internal
</div>
</div>
</div>
<!-- $(common.heading) -->

<!-- ^(home.row) -->
<div class="row d-flex flex-wrap">

<!-- ^(home.column) -->
<div class="col-md-6 col-lg-4 d-flex flex-column">

<!-- ^(common.card) -->
<div class="card d-flex flex-column flex-grow mt-3">

<!-- ^(common.card.body) -->
<div class="card-body d-flex flex-column">
<h5 class="card-title">
quux
</h5>
<p class="card-text flex-grow">Description of quux.</p>
</div>
<!-- $(common.card.body) -->

<!-- ^(common.card.footer) -->
<div class="card-footer text-muted text-truncate">
<small>Last activity: 15m ago<span class="text-danger"> (failed)</span></small>
</div>
<!-- $(common.card.footer) -->

</div>
<!-- $(common.card) -->

</div>
<!-- $(home.column) -->

<!-- ^(home.column) -->
<div class="col-md-6 col-lg-4 d-flex flex-column">

<!-- ^(common.card) -->
<div class="card d-flex flex-column flex-grow mt-3">

<!-- ^(common.card.body) -->
<div class="card-body d-flex flex-column">
<h5 class="card-title">
quuz
</h5>
<p class="card-text flex-grow">Description of quuz.</p>
</div>
<!-- $(common.card.body) -->

<!-- ^(common.card.footer) -->
<div class="card-footer text-muted text-truncate">
<small>Last activity: 30s ago<span class="text-success"> (just now)</span></small>
</div>
<!-- $(common.card.footer) -->

</div>
<!-- $(common.card) -->

</div>
<!-- $(home.column) -->

<!-- ^(home.column) -->
<div class="col-md-6 col-lg-4 d-flex flex-column">

<!-- ^(common.card) -->
<div class="card d-flex flex-column flex-grow mt-3">

<!-- ^(common.card.body) -->
<div class="card-body d-flex flex-column">
<h5 class="card-title">
qux
</h5>
<p class="card-text flex-grow">Description of qux.</p>
</div>
<!-- $(common.card.body) -->

<!-- ^(common.card.footer) -->
<div class="card-footer text-muted text-truncate">
<small>Last activity: 2d ago</small>
</div>
<!-- $(common.card.footer) -->

</div>
<!-- $(common.card) -->

</div>
<!-- $(home.column) -->

</div>
<!-- $(home.row) -->

</div>
<!-- $(home.internal) -->

</main>
<!-- $(base.main) -->

<!-- ^(base.footer) -->
<footer class="border-top">
<div class="container text-center">
<small class="text-muted">Last reloaded: 1985-10-26 06:02:30 PST</small>
</div>
</footer>
<!-- $(base.footer) -->

<script src="https://code.jquery.com/jquery-3.2.1.slim.min.js"
        integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN"
        crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js"
        integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q"
        crossorigin="anonymous"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.3/js/bootstrap.min.js"
        integrity="sha384-a5N7Y/aK3qNeh15eJKGWxsqtnX/wWdSZSKp+81YjTmS15nvnvxKHuzaWwXHDli+4"
        crossorigin="anonymous"></script>
<script>
$('.navbar-toggler').one('click', function(e) {
  setTimeout(function() {
    var height = $('header').height() + 1;
    var css = '<style> .anchor.offset { padding-top: %spx; margin-top: -%spx; } </style>';
    $(css.replace(/%s/g, height)).appendTo('head');
  }, 500);
});
$('.navbar-toggler').on('click', function(e) {
  $('.anchor').toggleClass('offset');
});
</script>


</body>

</html>